ext/rugged/rugged_remote.c

```diff
@@ -536,6 +553,10 @@ static VALUE rb_git_remote_fetch(int argc, VALUE *argv, VALUE self)
 		VALUE rb_val = rb_hash_aref(rb_options, CSTR2SYM("message"));
 		if (!NIL_P(rb_val))
 			log_message = StringValueCStr(rb_val);
+
+		VALUE rb_prune_type = rb_hash_aref(rb_options, CSTR2SYM("prune"));
+		if (!NIL_P(rb_prune_type))
```

Comment:
Either remove the `NIL_P` check here, or more preferably, remove the `rb_prune_type == Qnil` from inside `parse_prune_type`.